packages/astro/src/core/build/buildPipeline.ts

```diff
@@ -20,10 +18,6 @@ export class BuildPipeline extends Pipeline {
 	#internals: BuildInternals;
 	#staticBuildOptions: StaticBuildOptions;
 	#manifest: SSRManifest;
-	#currentEndpointBody?: {
-		body: string | Uint8Array;
-		encoding: BufferEncoding;
-	};
```

Comment:
One of the issues that forced me to keep this object is that we have a test that checks if we don't mangle the body or response, such as images. I

Reply:
Yeah I found that too (and CI is also erroring on that 😅), but I was able to make it not mangle for simple objects before. Currently it's erroring for `ResponseWithEncoding`. Thanks for the note!

Reply:
Aight I think I fixed it. I had to add the same encode logic inside `ResponseWithEncoding` so that it's on par with simple objects. I think that make sense for string responses, and otherwise the migration guide wouldn't be as nice, e.g. they have to do this themselves:

```js
new ResponseWithEncoding(Buffer.from('...', 'binary'), undefined, 'binary')
```